File: algorithms/Strings/Medium/LongestSubstringWithoutRepeatingCharacter/VSDLongestSubstringWithoutRepeatingCharacter.java
import java.util.HashMap;

public class Main{

    static int lengthOfLongestSubstring(String s){
        int start=0,end=0,ans=0;
        HashMap<Character,Integer> map = new HashMap<Character,Integer>();

        while(end<s.length()){
            if(!map.containsKey(end)){
                map.put(s.charAt(end), end);
                end++;
                ans=Math.max(map.size(),ans);
            }
            else{
                map.remove(s.charAt(start));
            }

        }
        return ans;
    }

    public static void main(String[] args) {
        String s = "abcabcbb";
        System.out.println(lengthOfLongestSubstring(s));
    }
}

File: algorithms/Strings/Medium/LongestSubstringWithoutRepeatingCharacter/VSDLongestSubstringWithoutRepeatingCharacter.py

def lengthOFLongestSubstring(string : str) -> int:

    start=end=ans=0
    hashset=dict()

    while end<len(string):
        if string[end] not in hashset.keys():
            hashset[string[end]]=end
            end+=1
            ans=max(len(hashset),ans)
        else:
            del hashset[string[start]]
            start+=1
    return ans 

if __name__=="__main__":
    s = "abcabcbb"
    print(lengthOFLongestSubstring(s))